src/lucky/renderable.cr

```diff
@@ -142,31 +142,43 @@ module Lucky::Renderable
     file(path, content_type, disposition, filename, status.value)
   end
 
+  private def send_response(body : String, content_type : String, status : Int32? = nil) : Lucky::TextResponse
+    Lucky::TextResponse.new(context, content_type, body, status: status)
+  end
+
   private def text(body : String, status : Int32? = nil) : Lucky::TextResponse
-    Lucky::TextResponse.new(context, "text/plain", body, status: status)
+    send_response(body, "text/plain", status)
   end
 
   private def text(body : String, status : HTTP::Status) : Lucky::TextResponse
-    Lucky::TextResponse.new(context, "text/plain", body, status: status.value)
+    text(body, status: status.value)
   end
 
   private def render_text(*args, **named_args) : Lucky::TextResponse
     text(*args, **named_args)
   end
 
   private def head(status : Int32) : Lucky::TextResponse
-    Lucky::TextResponse.new(context, content_type: "", body: "", status: status)
+    send_response(body: "", content_type: "", status: status)
   end
 
   private def head(status : HTTP::Status) : Lucky::TextResponse
     head(status.value)
   end
 
   private def json(body, status : Int32? = nil) : Lucky::TextResponse
-    Lucky::TextResponse.new(context, "application/json", body.to_json, status)
+    send_response(body.to_json, "application/json", status)
+  end
+
+  private def json(body, status : HTTP::Status) : Lucky::TextResponse
+    json(body, status: status.value)
```

Comment:
It looks like this one does not call `to_json` on the body, but probably should. Maybe there is a missing spec for it?

Reply:
since `json` isn't typed, it just passes the `body` to the previous def `json` which calls `to_json`. If we call it here too, then it would `to_json.to_json`.